### PR TITLE
4.0.0

### PR DESCRIPTION
**Pull request by ParadoxOrigins**
Initial pull to review before official release owo
@Niputi if it isn't too much can you check things like usage.md to see if i didn't made typos and such?

### CODE REVIEW

File: usage.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+* `%USER%` => Will be replaced by the mention of the user who just joined, like `@Baguette`. This tag isn't available for the farewell message
+* `%USERNAME%` => Will be replaced by the username of the user who just joined, like `Baguette`
+* `%USERTAG%` => Will be replaced by the username#discriminator of the user who just joined, like `Baguette#0001`
+* `%GUILD%` => Will be replaced by the name of the server
+* `%MEMBERCOUNT%` => Will be replaced by the new count of members in this server
+
+### Greetings and farewell target 
+
+You have a full control of where the greetings messages are sent, you can set it so they will be sent to a specific channel, or directly to the member who just joined.
```
Comment (Niputi) [inline usage.md]: `leave` => `leaves`

File: usage.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+#### Level up tags
+
+When setting up your own level up messages, you can use a few tags which will be replaced according to the context, here's the list of the tags you can use:
+
+* `%USER%` => Will be replaced by the mention of the user who just levelled up, like `@Baguette`
+* `%USERNAME%` => Will be replaced by the username of the user who just levelled up, like `Baguette`
+* `%USERTAG%` => Will be replaced by the username#discriminator of the user who just levelled up, like `Baguette#0001`
+* `%LEVEL%` => Will be replaced by the level the user just reached
+* `%WONROLES%` => This is a bit more complicated, the `%WONROLES%` tag will be replaced by `and won the role(s) <role list>`, unless no roles has been won, then it is replaced by nothing. Let's take an example:
```
Comment (Niputi) [inline usage.md]: `has been won` => `have been won`

File: usage.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+Let's say our custom level up message is set to: `Hey %USER% you just levelled up to level %LEVEL% %WONROLES% !`
+
+Now, let's say we set the roles `baguette` and `croissant` to be given at level 2, when the user reach the level 2, it will look like this:
```
Comment (Niputi) [inline usage.md]: `when the user reach the level 2` => `when user reaches level 2`

File: usage.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+> Hey @Baguette you just levelled up to level 2 and won the role(s) `baguette`, `croissant` !
+
+And when the user reach the level 3, at which we didn't set any roles to be given, it will look like this:
```
Comment (Niputi) [inline usage.md]: `when the user reach the level 3` => `when the user reaches level 3`

File: usage.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+- * [Terms references](#terms-references)
+- * [Category permissions and command permissions](#category-permissions-and-command-permissions)
+- * [Permissions priority and inheritance](#permissions-priority-and-inheritance)
+- * [Special case for roles permissions groups](#special-case-for-roles-permissions-groups)
+- * [Additional notes](#additional-notes)
+
+## Commands arguments syntax
+
+For consistency purposes, as well as to make it easy to learn, all commands that takes multiple mandatory arguments can be called without specifying any arguments, you will 
+then be prompted each argument.
```
Comment (Niputi) [inline usage.md]: `be prompted each` => `be prompted for each argument`

**Comment by ParadoxOrigins** [comment]
kthx luv ya

**Comment by Niputi** [comment]
Your welcome. Love you too
________________________________

**Pull request merged**